File: ML/Usuario.cs
using System.ComponentModel.DataAnnotations;

namespace ML
{
    public class Usuario
    {
        //Propiedades
        public int? IdUsuario { get; set; }
        public string? UserName { get; set; }
        public string? Nombre { get; set; }
        public string? ApellidoPaterno { get; set; }
        public string? ApellidoMaterno { get; set; }
        public string? Email { get; set; } 
        public string? Passwrd { get; set; }
        public string? Sexo { get; set; }
        public string? Telefono { get; set; }
        public string? Celular { get; set; }
        public string? FechaNacimiento { get; set; }
        public string? CURP { get; set; }
        public Rol? Rol { get; set; } //Propiedad de navegación.
        public List<object>? Usuarios { get; set; }
    }
}

File: maurodmatangularapp.client/src/Models/Usuario.ts

import { Rol } from '../Models/Rol'

//Clase donde coloco mis propiedades.

export interface Usuario {
  idUsuario: number
  userName: string
  nombre: string
  apellidoPaterno: string
  apellidoMaterno: string
  email: string
  passwrd: string
  sexo: string
  telefono: string
  celular: string
  fechaNacimiento: string
  curp: string
  rol: Rol
  usuarios: any
}
